client/signup/config/flows.js

```diff
@@ -56,7 +56,10 @@ function getRedirectDestination( dependencies ) {
 }
 
 function getSignupDestination( dependencies ) {
-	return `/home/${ dependencies.siteSlug }`;
+	if ( dependencies?.siteSlug !== 'no-site' ) {
+		return `/home/${ dependencies.siteSlug }`;
+	}
+	return `/checkout/thank-you/no-site`;
```

Comment:
I am not sure if this change is necessary. Signup destination is the page where we take the user finally after payment is complete, or site has been created in the case of a free plan. In the case of siteless checkout, we still want to take the user to /home after payment is successful and not a thank you page. Could you please explain more about this change?

Reply:
This is actually one of the last things that I needed to figure out, the URLs generated here are using the `dependencies` object, however we'll have a proper `siteSlug` until the checkout is completed, so another mechanism will be probably needed to redirect the user to their site home.